src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -749,4 +776,38 @@ public override IReadOnlyList<Type> GetParameterTypes(string methodName)
 
         return null;
     }
+
+    private static Activity? CreateActivity(IServiceProvider serviceProvider, string methodName)
+    {
+        if (serviceProvider.GetService<ActivitySource>() is ActivitySource activitySource
+            && activitySource.HasListeners())
```

Comment:
Feels like SignalR should have its own activity source?

Reply:
+1

Reply:
The name should probably have "server" in it somewhere. There might be a SignalR client activity source in the future, and they'd have different names.